sfcasts/embedded-overlay.md

```diff
@@ -1,112 +1,54 @@
 
-Let's add a `console.log(response)` before the `response.ok` check. In the Console
-tab we can seet there's a `response.redirected` set to `true`,
-Let's add one more if, and `if (response.redirected === true)`, 
-we'll redirect the user to the login page with
-`window.location.href = response.url`.
+## Debugging for Non-authenticated Users
+
+*But* there's a hiccup for non-authenticated users. If we log out, add a product to the cart, and try to checkout again... *nothing happens*. If you open the Chrome Dev Tools, you can see that the request is redirected to a login page first, but our JavaScript logic doesn't follow that redirect. Let's fix that!
 
-and add the current pathname to the URL. If the user isn't authenticated,
-we'll do `Promise.reject()` saying "User is not authenticated!". I will also
-add a comment above.
+In our code, add a `console.log(response)` before the `response.ok` check. Back on our site, in the "Console" tab, we can see that `response.redirected` is set to `true`. Let's add one more `if` - `if (response.redirected === true)` - and we'll redirect the user to the login page with `window.location.href = response.url`. If the user *isn't* authenticated, we'll add `Promise.reject()` which will tell us that the `User is not authenticated!`. I'll also add a quick comment above.
 
 ## Redirecting Users Back to the Cart Page
 
-Now if we click on the checkout button - we're redirected
-to the login page. And if I login, Now, when users sign in, they're redirected
-to the home page instead of the cart page.
-Let's fix that too. In the LemonSqueezy Stimulus controller, after
-`response.url`, we'll add `?_target_path=` string and concatenate
-`window.location.pathname`.
-
-To make this actually work, we'll open `src/Security/LoginFormAuthenticator`,
-and at the start of the `onAuthenticationSuccess` method
-add `if ($targetPath = $request->query->get())` passing out `_target_path`
-from the query parameters. Then `return new RedirectResponse($targetPath)`.
-
-Now, when users sign in after being redirected to the login page, they're
-taken straight back to the cart page. If they click the checkout button
-again, they'll see our awesome checkout overlay. And after completing the
-checkout - success! Press continue - here's our success message. All good!
-
-Next, let's see how we can listen to the LemonSqueezy JavaScript events that
-will also allow us to sync customer ID with the current user.
+Okay, *now* if we click the checkout button when we're not logged in... we're redirected to the login page! Nice! And if we enter our info and log in... we're redirected to the home page instead of the cart page. Let's fix that too! In `lemon-squeezy_controller.js`, after `response.url`, add a `?_target_path=` string, and concatenate `window.location.pathname`.
```

Comment:
```suggestion
Okay, *now* if we click the checkout button when we're not logged in... we're redirected to the login page! Nice! And if we enter our credentials and log in... we're redirected to the home page instead of the cart page. Let's fix that too! In `lemon-squeezy_controller.js`, after `response.url`, add a `?_target_path=` string, and concatenate `window.location.pathname`.
```